azure-pipelines.yml

```diff
@@ -312,27 +315,73 @@ stages:
               sqlcmd -U sa -P "ReallyStrongPwd1234%^&*" -Q "CREATE DATABASE test_ci;" -o create_db_output.txt
 
           - script: |
+              pip install --requirement requirements-dev-util.txt --requirement requirements-dev-test.txt --requirement requirements-dev-sqlalchemy.txt
               pip install --requirement requirements.txt
-              pip install --requirement requirements-dev-test.txt --requirement requirements-dev-sqlalchemy.txt --requirement requirements-dev-util.txt
               pip install .
             displayName: 'Install dependencies'
 
           - script: |
               pip install pytest pytest-cov pytest-azurepipelines
-              pytest --mssql --no-postgresql --no-spark --napoleon-docstrings --junitxml=junit/test-results.xml --cov=. --cov-report=xml --cov-report=html
+              pytest --mssql --no-postgresql --no-spark --napoleon-docstrings --junitxml=junit/test-results.xml --cov=. --cov-report=xml --cov-report=html --ignore=tests/cli --ignore=tests/integration/usage_statistics
+            displayName: 'pytest'
+
+  - stage: cli_integration
+    dependsOn: [scope_check, lint, usage_stats_integration, required, db_integration]
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    jobs:
+      - job: test_cli
+        condition: or(eq(stageDependencies.scope_check.changes.outputs['CheckChanges.GEChanged'], true), eq(variables.isMain, true))
+
+        services:
+          postgres: postgres
+
+        variables:
+          python.version: '3.8'
+
+        steps:
+          - task: UsePythonVersion@0
+            inputs:
+              versionSpec: '$(python.version)'
+            displayName: 'Use Python $(python.version)'
+
+          - bash: python -m pip install --upgrade pip==20.2.4
+            displayName: 'Update pip'
+
+          - script: |
+              sudo apt-get install -y pandoc
+              pip install pypandoc
+            displayName: 'Install pandoc'
+
+          - script: |
+              # not including "--requirement requirements-dev-spark.txt", because it specifies "pyspark>=2.3.2,<3.0.0" versions, which do not support Python 3.8 (see script step below)
```

Comment:
Don't we have support for pyspark 3 now? Can we change requirements-dev?

Reply:
@jcampbell I was *hoping* that we could change `requirements-dev-spark.txt` to use `Spark 3.x`.  However, I was a little unease about it for two reasons:
* Has everybody on the Engineering Team at Superconductive already standardized on `Spark 3.x`?  And does it work smoothly for everybody in terms of S3 access and any other relevant integrations?
* Are there companies still using `Spark 2.4`?  I recall that as of a year ago, companies were upgrading from `Spark 2.3` to `Spark 2.4` and not yet on `Spark 3.x`. 
* IMPORTANT: `Spark 3.x` does *NOT* support `Python 3.8` -- this might present difficulties to companies using `Python 3.7` and `Python 3.6` (validate those `Python` versions in our compatibility testing). 

On the basis of this reasoning, I felt that it would be best to defer the global requirement change to a future task.

Thank you.  /cc @eugmandel